src/parser_urdf_TEST.cc

```diff
@@ -994,6 +995,1096 @@ TEST(URDFParser, ParseWhitespace)
+        <link name='link2'/>
+        <joint name='joint1_2' type='fixed'>
+          <parent link='link1' />
+          <child  link='link2' />
+          <origin xyz='1 2 3' rpy='0.0 0.0 1.57'/>
+        </joint>
+        <link name='link3'>
+          <inertial>
+            <mass value='0.1' />
+            <origin rpy='1.570796326794895 0 0' xyz='0.123456789123456 0 0.0' />
+            <inertia ixx='0.01' ixy='0' ixz='0' iyy='0.01' iyz='0' izz='0.01' />
+          </inertial>
+        </link>
+        <joint name='joint2_3' type='continuous'>
+          <parent link='link2' />
+          <child  link='link3' />
+          <origin xyz='0.0 0.0 0.0' rpy='0.0 0.0 1.57'/>
+        </joint>
+        <gazebo reference='joint1_2'>
+          <disableFixedJointLumping>true</disableFixedJointLumping>
+        </gazebo>
+        <gazebo reference='joint1_2'>
+          <preserveFixedJoint>true</preserveFixedJoint>
+        </gazebo>
+      </robot>)";
+
+    sdf::URDF2SDF parser;
+    sdf::ParserConfig config;
+    config.URDFSetPreserveFixedJoint(true);
+    tinyxml2::XMLDocument sdfResult;
+    parser.InitModelString(str, config, &sdfResult);
+
+    // link2 will be converted to a frame, relative to link1, joint1_2 will be
+    // left out
+    EXPECT_PRED2(sdf::testing::notContains, buffer.str(),
+        "urdf2sdf: link[link2] has no <inertial> block defined, but does not "
+        "have a fixed parent joint, unable to be converted into a frame in "
+        "sdf");
+
+    tinyxml2::XMLElement *sdf = sdfResult.FirstChildElement("sdf");
+    ASSERT_NE(nullptr, sdf);
+    tinyxml2::XMLElement *model = sdf->FirstChildElement("model");
+    ASSERT_NE(nullptr, model);
+
+    tinyxml2::XMLElement *link = model->FirstChildElement("link");
+    ASSERT_NE(nullptr, link);
+    ASSERT_NE(nullptr, link->Attribute("name"));
+    EXPECT_EQ("link1", std::string(link->Attribute("name")));
+
+    tinyxml2::XMLElement *frame = model->FirstChildElement("frame");
+    ASSERT_NE(nullptr, frame);
+    ASSERT_NE(nullptr, frame->Attribute("name"));
+    EXPECT_EQ("link2", std::string(frame->Attribute("name")));
+    ASSERT_NE(nullptr, frame->Attribute("attached_to"));
+    EXPECT_EQ("link1", std::string(frame->Attribute("attached_to")));
+    tinyxml2::XMLElement *framePose = frame->FirstChildElement("pose");
+    ASSERT_NE(nullptr, framePose);
+    ASSERT_NE(nullptr, framePose->Attribute("relative_to"));
+    EXPECT_EQ("link1", std::string(framePose->Attribute("relative_to")));
+
+    tinyxml2::XMLElement *joint = model->FirstChildElement("joint");
+    ASSERT_NE(nullptr, joint);
+    ASSERT_NE(nullptr, joint->Attribute("name"));
+    EXPECT_EQ("joint2_3", std::string(joint->Attribute("name")));
+    joint = joint->NextSiblingElement("joint");
+    EXPECT_EQ(nullptr, joint);
+
+    link = link->NextSiblingElement("link");
+    ASSERT_NE(nullptr, link);
+    ASSERT_NE(nullptr, link->Attribute("name"));
+    EXPECT_EQ("link3", std::string(link->Attribute("name")));
```

Comment:
Can you check that the generated SDF is valid? You can load it via `sdf::Root` and check if there are any errors.

Reply:
I have all the testing with `sdf::Root` in `test/integration/urdf_to_sdf.cc`, https://github.com/gazebosim/sdformat/pull/1238/files#diff-ba763ab70ecd0b5b139acc603e4a4f2861453b0480d80ae5ff9e6d8ed6d4398cR146